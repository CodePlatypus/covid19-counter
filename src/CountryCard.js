import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import makeStyles from "@material-ui/core/styles/makeStyles";

const useStyles = makeStyles({
    root: {
        marginBottom: 10
    }
});

export default function CountryCard(props) {
    const classes = useStyles();
    const {
        country,
        cases,
        todayCases,
        deaths,
        todayDeaths,
        recovered,
        // activeCases,
        critical,
    } = props.countryStat;
    return (
        <Card className={classes.root}>
            <CardContent>
                <Typography variant="h5" component="h2">
                    {country}
                </Typography>
                <Typography color="textSecondary">
                    New Cases: {todayCases}
                </Typography>
                <Typography color="textSecondary">
                    New Deaths: {todayDeaths}
                </Typography>
                <Typography color="textSecondary">
                    Critical: {critical}
                </Typography>
                <Typography color="textSecondary">
                    Total Cases: {cases}
                </Typography>
                <Typography color="textSecondary">
                    Total Deaths: {deaths}
                </Typography>
                <Typography color="textSecondary">
                    Recovered: {recovered}
                </Typography>
            </CardContent>
        </Card>
    )
}

CountryCard.prototype = {
    countryStat: PropTypes.object.isRequired
};