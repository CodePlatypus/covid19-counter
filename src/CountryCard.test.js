import React from 'react';
import {shallow} from 'enzyme';
import CountryCard from './CountryCard';
import toJson from 'enzyme-to-json';

test('renders CardBody with countryStat', () => {
    const wrapper = shallow(<CountryCard countryStat={mockData} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

const mockData = {
    "country": "China",
    "cases": 80967,
    "todayCases": 39,
    "deaths": 3248,
    "todayDeaths": 3,
    "recovered": 71150,
    "activeCases": 6569,
    "critical": 2136
};