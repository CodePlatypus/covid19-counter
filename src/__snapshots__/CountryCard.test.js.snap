// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardBody with countryStat 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="h2"
      variant="h5"
    >
      China
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
    >
      New Cases: 
      39
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
    >
      New Deaths: 
      3
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
    >
      Active Cases: 
      6569
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
    >
      Critical: 
      2136
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
    >
      Total Cases: 
      80967
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
    >
      Total Deaths: 
      3248
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      color="textSecondary"
    >
      Recovered: 
      71150
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
